core/src/main/java/com/yahoo/imapnio/async/data/SearchResult.java

```diff
@@ -6,20 +6,42 @@
 import javax.annotation.Nullable;
 
 /**
- * This class provides the list of message sequence numbers from search command response.
+ * This class provides the modification sequence and the list of message sequence numbers from search command response.
  */
 public class SearchResult {
     /** Search command response sequence number, could be message sequence or UID. */
-    @Nonnull
     private final List<Long> msgNumbers;
```

Comment:
why remove nonnull ?  that will not be backward compatible, lets discuss

Reply:
The getter msgNumbers  of is Nullable, but I will revert since it should not be included in this PR.